### PR TITLE
travisci_automatic_releases

### PR DESCRIPTION
**Pull request by goru97**
**What:**
Changed .travis.yml

**Why:**
To create UberJar (Earlier it was mvn install, now it's mvn package -P all-modules ).
     a. So that it can be used directly whenever we make official releases.
     b. So that it can be used by Docker image.

**How:**
Will create a wiki for that.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,6 +18,26 @@ script:
...
+  - git push -q https://$API_KEY@github.com/rackerlabs/blueflood --tags
+
+deploy:
+  skip_cleanup: true
+  provider: releases
+  api_key: 
+    secure: Eue8cRnapBQim2wArFkR88r+SJHe4Qs5gcor8QCWJuFltJcRG7JCu+y7FAjeYSCBqXGNVENirRjV5YnfdQS4uNKhb3m3WalneJDGCUtk82VdeJiNwV52IjKJEmpTUQu+bun1aoU0+ZY2PkzFW8S1lpCzj+n/YH3LWgIiXayiOhOzL2Yq1VJp2/BVnEZvKRpVlN55SwFPTkWVta2qKpr0b7KfNpPLQArQqJUExrHMQxFb2iM1Yb0xp2C2HQIYSr3xYw9eR5TjFI2NEZwfzhsrJ5M/ZVbkM5UAOranHGiQvJ2Eha8o0XHqvjOCPcJUVXJ7M9+xRA5THMVBrLCMZOH9LIskwHQxx9318b1jfw2uM+fwEZwqfyakCN/pmaQFxzzBmVDsFwlkoqzYIRimanOlLhwyNY9Nkc0XN40ksvIdmIoG0llOtZFQf168WsWUT5GR1QQrNe1sAgWdPQ51suxEcL0HITwlna4VUakdbNxBZjZ5+//pTMohH5aWlkJSt+4oJIsq0tUpDonGQ7pJ9iR12cISiXmjBAEFDKtPM2JJsBuP73kLmgHS11lUY0sFtz5oyR9NjBr2Gjrs15XCFqZwn21SiTzunHo9e3k2oiU74g3B7UZ+YiiKRhnv5gTu7Gps4MgpVlIsOPU4h9ygZQJga2odoJKKMMFQlyCsGHyo54U=
+  file_glob: true
+  file:
+    - "./blueflood-all/target/blueflood-all-2.0.0-SNAPSHOT-jar-with-dependencies.jar"
```
Comment (tilogaat) [inline .travis.yml]: can we change the jar name, depending on the build version?

File: .travis.yml [hunk 2]
```diff
@@ -18,6 +18,26 @@ script:
...
+  - git config --global user.name "Travis CI"
+  - export GIT_TAG=$TRAVIS_BRANCH-v1.0.$TRAVIS_BUILD_NUMBER
+  - git tag $GIT_TAG -a -m "Generated tag from TravisCI for build $TRAVIS_BUILD_NUMBER"
+  - git push -q https://$API_KEY@github.com/rackerlabs/blueflood --tags
+
+deploy:
+  skip_cleanup: true
+  provider: releases
+  api_key: 
+    secure: Eue8cRnapBQim2wArFkR88r+SJHe4Qs5gcor8QCWJuFltJcRG7JCu+y7FAjeYSCBqXGNVENirRjV5YnfdQS4uNKhb3m3WalneJDGCUtk82VdeJiNwV52IjKJEmpTUQu+bun1aoU0+ZY2PkzFW8S1lpCzj+n/YH3LWgIiXayiOhOzL2Yq1VJp2/BVnEZvKRpVlN55SwFPTkWVta2qKpr0b7KfNpPLQArQqJUExrHMQxFb2iM1Yb0xp2C2HQIYSr3xYw9eR5TjFI2NEZwfzhsrJ5M/ZVbkM5UAOranHGiQvJ2Eha8o0XHqvjOCPcJUVXJ7M9+xRA5THMVBrLCMZOH9LIskwHQxx9318b1jfw2uM+fwEZwqfyakCN/pmaQFxzzBmVDsFwlkoqzYIRimanOlLhwyNY9Nkc0XN40ksvIdmIoG0llOtZFQf168WsWUT5GR1QQrNe1sAgWdPQ51suxEcL0HITwlna4VUakdbNxBZjZ5+//pTMohH5aWlkJSt+4oJIsq0tUpDonGQ7pJ9iR12cISiXmjBAEFDKtPM2JJsBuP73kLmgHS11lUY0sFtz5oyR9NjBr2Gjrs15XCFqZwn21SiTzunHo9e3k2oiU74g3B7UZ+YiiKRhnv5gTu7Gps4MgpVlIsOPU4h9ygZQJga2odoJKKMMFQlyCsGHyo54U=
```
Comment (tilogaat) [inline .travis.yml]: What's the best way to configure this key? I don't like it being there in the yaml file. Can it be there in blueflood.conf?

File: .travis.yml (hunk 1 above)
Comment (goru97) [inline .travis.yml]: that modification has to be done in maven's pom.xml. I am using travis-CI build number to tag our new release.

File: .travis.yml (hunk 2 above)
Comment (goru97) [inline .travis.yml]: This is the best way. The api key is encrypted by the travis CLI, which only works for git repo. and whosoever  has the access to that repo can trigger this push.

Comment (goru97) [inline .travis.yml]: also if you notice there is one env variable $API_KEY which is actually not encrypted like the one below, hence that one is saved on travis CI web and can only be changed by someone who can access the repo.

**Comment by tilogaat** [comment]
+1

**Pull request merged**